src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -135,6 +135,38 @@ private void OnDispatchedToUI(object sender, UIDispatcherEventArgs e)
         /// </summary>
         public event Action<PortModel> PortDisconnected;
 
+        public class NodeExecutionEventArgs : EventArgs
+        {
+            public Guid GUID { get; private set; }
+            public object Data { get; private set; }
+
+            public NodeExecutionEventArgs(NodeModel model, object data)
+            {
+                this.GUID = model.GUID;
+                this.Data = data;
+            }
+        }
+
+        /// <summary>
+        /// Event triggered before a node is executed.
```

Comment:
I think this comment should indicate that these events are only fired when profiling is enabled.

Reply:
alternatively we could rename them to be less general.

Reply:
Notes have been added to both events to indicate that they will only be fired when profiling is turned on. My thought is that the names should remain general in the case that it is decided to make the event firing non-dependent on profiling being turned on so to not break the API.